### DISCUSSION TITLE
Compare columns of different tables on a relation findMany

### DISCUSSION THREAD

**Discussion by miguelmagueijo** [prisma/prisma#15637] [Q&A] [answered]
Is there a way to compare two columns of different tables when doing a findMany with inner joins? (SQLITE)

**Example I have**:

I have 5 tables, User, Product, Shop, UserProductFollow, ProductShop.

- **User**: id, name
- **Product**: id, name
- **Shop**: id, name
- **UserProductFollow**: userId, productId, targetPrice
- **ProductShop**: productId, shopId, price.

**User** M - N **Product** (**Many-to-Many** through **UserProductFollow**)
**Product** M - N **Shop** (**Many-to-Many** through **ProductShop)**

I want to get all users ID and product ID that target price is greater than the price on the shop.

I know the Raw SQL but I cannot find a way to do it with Prisma.
`
SELECT u.id, p.id
FROM User AS u 
        INNER JOIN UserProductFollow AS upf ON u.id = upf.userId 
	INNER JOIN Product AS p ON upf.productId = p.id
	INNER JOIN ProductShop AS ps ON p.id = ps.productId
WHERE upf.price >= ps.price;
`

EDIT: Database being used is SQLITE

**Comment by austincrim** [comment, marked as answer]
Hey @miguelmagueijo 👋,

This is not yet possible in Prisma without [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw), but we have taken the first steps toward supporting it with the recent [`fieldReference`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#compare-columns-in-the-same-table) preview feature which allows you to compare different fields **of the same table**.

Feel free to offer any feedback on the current and potential design of the `fieldReference` feature in [this issue](https://github.com/prisma/prisma/issues/15068).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
